app-office/texmacs/texmacs-1.99.12.recipe

```diff
@@ -0,0 +1,100 @@
+SUMMARY="A scientific editing platform designed to create beautiful technical documents"
+DESCRIPTION="TeXmacs provides a unified and user friendly framework for editing \
+structured documents with different types of content: text, mathematics, graphics, \
+interactive content, slides, etc.\
+Its rendering engine uses high-quality typesetting algorithms so as to produce \
+professionally looking documents, which can either be printed out or presented from \
+a laptop."
+HOMEPAGE="http://www.texmacs.org/"
+COPYRIGHT="Free Software Foundation"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/mgubi/texmacs/archive/texmacs-$portVersion-haiku-qt5.tar.gz"
+CHECKSUM_SHA256="607fa4db4676b78780f7af086feec2e2332dc18a371d79fa50d640de6db10b43"
+SOURCE_DIR="texmacs-texmacs-$portVersion-haiku-qt5/src"
+PATCHES="texmacs-$portVersion.patchset"
+ADDITIONAL_FILES="texmacs.rdef.in"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	texmacs$secondaryArchSuffix = $portVersion
+	app:TeXmacs
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libQt5Concurrent$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Svg$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libguile$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libltdl$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libiconv$secondaryArchSuffix
+	devel:libQt5Concurrent$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Svg$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libguile$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:automake
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:gettext$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:sed
+	cmd:guile == 1.8.8
+	"
+
+BUILD()
+{
+	runConfigure ./configure
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	mkdir $appsDir
+	rm $binDir/texmacs
+	mv $binDir/fig2ps $libExecDir/TeXmacs/bin/fig2ps
+	mv $libExecDir/TeXmacs/bin/texmacs.bin $appsDir/TeXmacs
```

Comment:
Some suggestions, atm there is an empty $binDir, $libExecDir/TeXmacs/lib and we don't use pixmaps ($dataDir/pixmaps)
```diff
-	rm $binDir/texmacs
	mv $binDir/fig2ps $libExecDir/TeXmacs/bin/fig2ps
	mv $libExecDir/TeXmacs/bin/texmacs.bin $appsDir/TeXmacs
+	rm -rf $binDir
+	rm -rf $dataDir/pixmaps
+	rm -rf $libExecDir/TeXmacs/lib
```

Maybe we need to review $libExecDir also here, atm it points to $prefix/lib instead of $libDir: https://github.com/haikuports/haikuporter/blob/8d5c14b6e9d4d0b0477e798afee40281f194c923/HaikuPorter/Port.py#L1173
Search reveals a --omit-dirs for this also: https://github.com/haikuports/haikuports/search?q=%24libExecDir&unscoped_q=%24libExecDir

Reply:
Strike the suggestion for now, not doing what I had in mind :)

Reply:
- We remove the pixmaps directory: it contains the images for all the icons in TeXmacs, both in xpm and png format. They are needed for the program to function in any platform.
- I think now the libExecDir stuff is fine. If we try to change it from the configure line I will have to  patch the Makefile. I do not see any problem with the current recipe. TeXmacs install a certain number of binary files into $libexecdir/TeXmacs/bin and I just moved there fig2ps in order not to pollute #binDir unnecessarily.